src/modules/formBuilder/components/formTree/FormTree.tsx

```diff
@@ -1,34 +1,41 @@
 import { RichTreeView } from '@mui/x-tree-view/RichTreeView';
 import React, { useMemo } from 'react';
 import Loading from '@/components/elements/Loading';
-import FormTreeItem from '@/modules/formBuilder/components/formTree/FormTreeItem';
+import FormTreeItem, {
+  TreeItemContext,
+} from '@/modules/formBuilder/components/formTree/FormTreeItem';
 import { getItemsForTree } from '@/modules/formBuilder/components/formTree/formTreeUtil';
-import { FormDefinitionJson } from '@/types/gqlTypes';
+import { FormDefinitionJson, FormItem } from '@/types/gqlTypes';
 
 interface FormTreeProps {
   definition: FormDefinitionJson;
+  setSelectedItem: (item: FormItem) => void;
 }
-const FormTree: React.FC<FormTreeProps> = ({ definition }) => {
+const FormTree: React.FC<FormTreeProps> = ({ definition, setSelectedItem }) => {
   const definitionForTree = useMemo(
     () => getItemsForTree(definition),
     [definition]
   );
 
+  const context = React.useMemo(
+    () => ({
+      onEditButtonClicked: setSelectedItem,
+    }),
+    [setSelectedItem]
+  );
+
   if (!definition) return <Loading />;
 
   return (
-    <RichTreeView
-      aria-label='form tree view'
-      disableSelection
-      items={definitionForTree}
-      getItemId={(item) => item.linkId}
-      getItemLabel={(item) => item.text || item.helperText || item.linkId}
-      slots={{ item: FormTreeItem }}
-      onItemFocus={(_event, itemId) => {
-        // TODO - add node onclick handler
-        console.log(itemId);
-      }}
-    />
+    <TreeItemContext.Provider value={context}>
```

Comment:
(this provider gets around needing to pass props directly to the custom TreeItem element. Took this idea from the example here: https://mui.com/x/react-tree-view/rich-tree-view/customization/#custom-label)

Reply:
Nice. So far we usually use a pattern of creating the context in its own file, like `src/modules/auth/AuthContext.tsx` instead of inside a component file. I find that a bit neater, since the context itself isn't really tied to one component.